src/tree/heavylight2.cpp

```diff
@@ -1,16 +1,28 @@
+/**
+ * Author: Richard Hladik
+ * Source: The concept of HLD with rearranging of sons borrowed from <codeforces.com/blog/entry/53170>
+ * Description: Calculates the heavy-light decomposition of given tree.
+ *  Data for chains is stored in one flat structure.
+ *  Also usable for lowest common ancestor.
+ * Time: $O(n \log n)$
+ * Usage: initHLD();
+ *  // What is the value on edges on path from 1 to 2?
+ *  quepdate(1, 2, false, ZERO);
+ *  // Update path from 1 to 2 with value 30
+ *  quepdate(1, 2, true, 30);
+ */
 #include "../base.hpp"
 
-/* The concept of HLD with rearranging of sons borrowed from
- * <codeforces.com/blog/entry/53170> */
 vector<vector<int>> G;
 vector<int> et, in, out, subs, depth, top, par;
-
 typedef ll T;
 
-T ZERO = 0;
-T combine(T a, T b) { return a + b; }
+// Implement these four to customise behavior
+T ZERO = 0; // Neutral element for operations
+T combine(T a, T b) { return a + b; } // How to compose two results
+// Query or update interval from a to b (0<=a,b<|V|)
```

Comment:
The "(0<=a,b<|V|)" part needs more spaces, I needed some time to decipher it :-).